### ISSUE TITLE
SacreBLEU uses incorrect tokenizer for Japanese

### ISSUE THREAD

**Issue by ekoenitz** [huggingface/transformers#14758]
[SacreBLEU](https://github.com/mjpost/sacreBLEU) needs to be told to use a different tokenizer in order to properly evaluate Japanese text, but as far as I can tell the HuggingFace framework has no means with which to do this.  I imagine other languages that need special tokenizers are affected as well, but I haven't tested them.

## Environment info
- `transformers` version: 4.13.0.dev0
- Platform: Linux-5.4.0-84-generic-x86_64-with-glibc2.10
- Python version: 3.8.8
- PyTorch version (GPU?): 1.10.0+cu113 (True)
- Tensorflow version (GPU?): 2.6.0 (False)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help
- Text generation: @patrickvonplaten @narsil

## Information

Model I am using: mBART-50 and mT5

The problem arises when using:
* [ ] the official example scripts: /transformers/examples/pytorch/run_translation.py

The tasks I am working on is:
* [ ] my own task or dataset: [JESC](https://nlp.stanford.edu/projects/jesc/)

## To reproduce

1. Run a translation model as described in /transformers/examples/pytorch/README.md, with **Japanese as the target language**
2. Use the --do_eval, --do_predict, and --predict_with_generate flags
3. Note the low BLEU score
4. Run the [SacreBLEU](https://github.com/mjpost/sacreBLEU) command on detokenized target vs. generated_predictions.txt with Japanese tokenizer (example: sacrebleu dev_ja_detokenized.txt -i generated_predictions.txt --tokenize ja-mecab -b)
5. Note the different, probably higher, BLEU score

## Expected behavior

I would expect HuggingFace's SacreBLEU evaluation score to be the same as SacreBLEU proper.

To give an example of how big the gap gets: After 3 epochs of fine-tuning mT5-large on JESC, HF evaluated the BLEU score as only **5.17**.  Running the output through the SacreBLEU command, however, caused the score to jump to **11.90**, which is much more in-line with the ja-en score of 16.8.

**Comment by Narsil** [comment]
Hi @ekoenitz,

I am by no means an expert in this example.
However, looking at it, it seems you are adding an argument to help `sacrebleu` figuring out the japanese tokenizer `--tokenizer ja-meca` right ?

If that's the case, the bleu score is actually calculated by Sacrebleu too and you would need to send that option too., https://github.com/huggingface/transformers/blob/master/examples/pytorch/translation/run_translation.py#L515 Adding `metrics.compute(...., tokenizer="ja-meca")`  should behave the same as your command.

Could that be the explanation ?

If that's it, this could probably be added to the README.md ?

**Comment by ekoenitz** [comment]
Ah, I didn't know you could send metric-specific arguments like that.  That does solve the issue, yes.

Along with readme updates, perhaps adding something like an eval_tokenizer argument would be a good idea?  I was 50/50 on whether I should file this issue as a bug or a feature request.
